test/tests.transform.js

```diff
@@ -47,3 +47,57 @@ describe("_blockHoist nodes", function() {
     assert.deepEqual(names, ["hoistMe", "doNotHoistMe", "oyez"]);
   });
 });
+
+describe("marked", function() {
+  it("should work with a single function", function() {
+    var ast = recast.parse('function* foo(){};', {
+        parser: require("babylon")
+    });
+    const code = recast.print(transform(ast)).code;
+    assert.notStrictEqual(
+      code.indexOf(
+        'var _marked = regeneratorRuntime.mark(foo);'
+      ), -1
+    );
+  });
+  it("should work with multiple functions", function() {
+    var ast = recast.parse([
+      'function* foo(){};',
+      'function* bar() {};'
+    ].join("\n"), {
+        parser: require("babylon")
+    });
+    const code = recast.print(transform(ast)).code;
+    assert.notStrictEqual(
+      code.indexOf(
+        'var _marked = regeneratorRuntime.mark(foo), _marked2 = regeneratorRuntime.mark(bar);'
+      ), -1
+    );
+  });
+});
+
+// TODO - how do i get recast to return comments?
```

Comment:
i aint sure how what u are trying to achieve here exactly, ive managed to access comments like this:
```
recast.parse('var _marked = /*#__PURE__*/regeneratorRuntime.mark(test)').program.body[0].declarations[0].init.comments
```
maybe that helps you somehow